src/set_timeout.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2022 Collabora Limited
+# Author: Jeny Sadadia <jeny.sadadia@collabora.com>
+
+import logging
+import os
+import sys
+from datetime import datetime, timedelta
+from time import sleep
+
+import kernelci
+import kernelci.config
+import kernelci.db
+from kernelci.cli import Args, Command, parse_opts
+
+from logger import Logger
+
+
+class SetTimeout:
+
+    def __init__(self, configs, args):
+        self._db_config = configs['db_configs'][args.db_config]
+        api_token = os.getenv('API_TOKEN')
+        self._db = kernelci.db.get_db(self._db_config, api_token)
+        self._logger = Logger("config/logger.conf", "set_timeout")
+
+    def _set_timeout_status(self, node):
+        """Set Node status to timeout if maximum wait time is over"""
+        current_time = datetime.utcnow()
+        max_wait_time = datetime.strptime(node['created'],
+                                          '%Y-%m-%dT%H:%M:%S.%f') + \
```

Comment:
I thought the `created` time was already in ISO format? See how the `complete_hack.py` script uses that.